flink-core/src/main/java/org/apache/flink/configuration/description/HtmlFormatter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration.description;
+
+/**
+ * Formatter that transforms {@link Description} into Html representation.
+ */
+public class HtmlFormatter extends Formatter {
+
+	@Override
+	protected void formatLink(StringBuilder state, String link, String description) {
+		state.append(String.format("<a href=\"%s\">%s</a>", link, description));
+	}
+
+	@Override
+	protected void formatLineBreak(StringBuilder state) {
+		state.append("<br/>");
+	}
+
+	@Override
+	protected void formatText(StringBuilder state, String format, String[] elements) {
+		String escapedFormat = escapeCharacters(format);
+		state.append(String.format(escapedFormat, elements));
+	}
+
+	@Override
+	protected void formatList(StringBuilder state, String[] entries) {
+		state.append("<ul>");
+		for (String entry : entries) {
+			state.append(String.format("<li>%s</li>", entry));
+		}
+		state.append("</ul>");
+	}
+
+	@Override
+	protected Formatter newInstance() {
+		return new HtmlFormatter();
+	}
+
+	private static final String TEMPORARY_PLACEHOLDER = "superRandomTemporaryPlaceholder";
```

Comment:
this should be distinct from the placeholder in `Utils`.

Reply:
Does it matter? Technically it does not go out of this escaping.

Reply:
any my point is exactly to prevent bugs from masking each other. The placeholder was safe to use since we could be reasonably sure it will not be present in the input. Once 2 entities use the same placeholder, with access to the same input, this is no longer the case.